### ISSUE TITLE
Should move code out of module profile/startup scripts

### ISSUE THREAD

**Issue by Jaykul** [Azure/azure-powershell#3326]
All of [them](https://github.com/Azure/azure-powershell/search?utf8=%E2%9C%93&q=OnImport+ps1+Invoke&type=Code). 

For example, the AzureRM.Profile module implements [IModuleAssemblyInitializer](https://msdn.microsoft.com/en-us/library/system.management.automation.imoduleassemblyinitializer) in [such a way](https://github.com/Azure/azure-powershell/blob/7ccdbeef0a05c270ec20d73bd82375c13ac2505b/src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs#L254-L269) that it runs a `AzureRmProfileStartup.ps1` script which must be in the folder with the module... 

The current contents of the script just set command aliases -- in the most convoluted way imaginable. These obviously should be set in the `AzureRm.Profile.psm1` file, not in a Startup script which has to be invoked from the assembly.

```posh
New-Alias Login-AzureRmAccount        Add-AzureRmAccount -Description "AzureAlias"
New-Alias Select-AzureRmSubscription  Set-AzureRmContext -Description "AzureAlias"
```

The bottom line is that all of the modules should use a psm1 module rather than a profile startup script.

**Comment by markcowl** [comment]
@Jaykul This is specifically designed to allow using the aliases across thew widest set of ExecutionPolicies on a user machine.  Closing as by design.
